app/Services/Services/Sharing/ServiceOptionImporterService.php

```diff
@@ -0,0 +1,123 @@
+use Pterodactyl\Contracts\Repository\ServiceVariableRepositoryInterface;
+use Pterodactyl\Exceptions\Service\ServiceOption\DuplicateOptionTagException;
+
+class ServiceOptionImporterService
+{
+    /**
+     * @var \Illuminate\Database\ConnectionInterface
+     */
+    protected $connection;
+
+    /**
+     * @var \Pterodactyl\Contracts\Repository\ServiceOptionRepositoryInterface
+     */
+    protected $repository;
+
+    /**
+     * @var \Pterodactyl\Contracts\Repository\ServiceRepositoryInterface
+     */
+    protected $serviceRepository;
+
+    /**
+     * @var \Pterodactyl\Contracts\Repository\ServiceVariableRepositoryInterface
+     */
+    protected $serviceVariableRepository;
+
+    /**
+     * XMLImporterService constructor.
+     *
+     * @param \Illuminate\Database\ConnectionInterface                             $connection
+     * @param \Pterodactyl\Contracts\Repository\ServiceRepositoryInterface         $serviceRepository
+     * @param \Pterodactyl\Contracts\Repository\ServiceOptionRepositoryInterface   $repository
+     * @param \Pterodactyl\Contracts\Repository\ServiceVariableRepositoryInterface $serviceVariableRepository
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        ServiceRepositoryInterface $serviceRepository,
+        ServiceOptionRepositoryInterface $repository,
+        ServiceVariableRepositoryInterface $serviceVariableRepository
+    ) {
+        $this->connection = $connection;
+        $this->repository = $repository;
+        $this->serviceRepository = $serviceRepository;
+        $this->serviceVariableRepository = $serviceVariableRepository;
+    }
+
+    /**
+     * Take an uploaded XML file and parse it into a new service option.
+     *
+     * @param \Illuminate\Http\UploadedFile $file
+     * @param int                           $service
+     * @return \Pterodactyl\Models\ServiceOption
+     *
+     * @throws \Pterodactyl\Exceptions\Model\DataValidationException
+     * @throws \Pterodactyl\Exceptions\Repository\RecordNotFoundException
+     * @throws \Pterodactyl\Exceptions\Service\Pack\InvalidFileUploadException
+     */
+    public function handle(UploadedFile $file, int $service): ServiceOption
+    {
+        if (! $file->isValid() || ! $file->isFile()) {
+            throw new InvalidFileUploadException(trans('exceptions.service.exporter.import_file_error'));
+        }
+
+        $parsed = json_decode($file->openFile()->fread($file->getSize()));
+
+        if (object_get($parsed, 'meta.version') !== 'PTDL_v1') {
+            throw new InvalidFileUploadException(trans('exceptions.service.exporter.invalid_json_provided'));
+        }
+
+        $service = $this->serviceRepository->getWithOptions($service);
+        $service->options->each(function ($option) use ($parsed) {
+            if ($option->tag === object_get($parsed, 'tag')) {
+                throw new DuplicateOptionTagException(trans('exceptions.service.options.duplicate_tag'));
```

Comment:
It can probably just be removed entirely as we don't _need_ tags anymore since references in the daemon don't use symlinking anymore, and are just UUID based.

Reply:
oh so you want to use uuids within the panel and the email:tag for sharing only?